### PR TITLE
Image flavors

### PR DESCRIPTION
**Pull request by jarpy**
Introduce the "image flavour" system from elasticsearch-docker.

Unlike elasticsearch-docker, which requires three flavours for different licence levels, kibana-docker only requires `oss` and `x-pack` flavours.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,29 +1,47 @@
...
+
+VERSIONED_IMAGE := $(ELASTIC_REGISTRY)/kibana/kibana:$(VERSION_TAG)
+
+FIGLET := pyfiglet -w 160 -f puffy
+
+all: build test
+
+test: lint docker-compose
+	$(foreach FLAVOR, $(IMAGE_FLAVORS), \
+	  $(FIGLET) "test: $(ELASTIC_VERSION)-$(FLAVOR)"; \
```
Comment (dliappis) [inline Makefile]: Nice to see the use of `pyfiglet` expanding!

File: Makefile [hunk 2]
```diff
@@ -48,18 +72,25 @@ venv: requirements.txt
...
-
-# Generate docker-compose.yml from a Jinja2 template.
-docker-compose.yml: venv
-	jinja2 \
-	  -D version_tag='$(VERSION_TAG)' \
-	  templates/docker-compose.yml.j2 > docker-compose.yml
-
-.PHONY: build clean flake8 push pytest test dockerfile docker-compose.yml
+# Generate the Dockerfiles from Jinja2 templates.
+dockerfile: venv templates/Dockerfile*.j2
```
Comment (dliappis) [inline Makefile]: While here `*` is used, later on only `templates/Dockerfile.j2` seems to be referenced; are there multiple Dockerfile jinja2 templates?

File: bin/elastic-version [hunk 3]
```diff
@@ -6,18 +6,45 @@
...
+    try:
+        return version_map['branches'][branch]
+    except KeyError:
+        # Unknown branch, assume it is a fork of master.
+        return version_map['branches']['master']
+
 
 if __name__ == '__main__':
-    print(elastic_version)
+    if 'ELASTIC_VERSION' in environ:
```
Comment (dliappis) [inline bin/elastic-version]: Thanks for making this up to date with the code in elasticsearch-docker!

File: templates/docker-compose.yml.j2 [hunk 4]
```diff
@@ -2,10 +2,10 @@
 version: '3'
 services:
   kibana:
-    image: docker.elastic.co/kibana/kibana:{{ version_tag }}
+    image: docker.elastic.co/kibana/kibana:{{ version_tag }}-{{ image_flavor }}
     ports:
       - 5601:5601
-    networks: [elastic]
+    networks: [elastic]
```
Comment (dliappis) [inline templates/docker-compose.yml.j2]: Some stray trailing whitespace characters here, could be trimmed.

File: Makefile (hunk 2 above)
Comment (jarpy) [inline Makefile]: Legacy from an earlier implementation. Good catch!

File: templates/docker-compose.yml.j2 (hunk 4 above)
Comment (jarpy) [inline templates/docker-compose.yml.j2]: Thanks. I had it open in fundamental mode (eww), so no whitespace-mode.

File: bin/elastic-version (hunk 3 above)
Comment (jarpy) [inline bin/elastic-version]: No problem. That cut-n-paste is breaking my heart though.

**Pull request merged**